Bottom: 63d3bf9e2dfd8125193799f1932cfb1a832edae3
Top:    3e94644bc5fa43a6e5255afdf819f644ce1cf37f
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-02-21 13:37:00 +0800

Refresh of retry

---

diff --git a/samples/bpf/xdp_tx_iptunnel_kern.c b/samples/bpf/xdp_tx_iptunnel_kern.c
index 0f4f6e8..cd2d2bc 100644
--- a/samples/bpf/xdp_tx_iptunnel_kern.c
+++ b/samples/bpf/xdp_tx_iptunnel_kern.c
@@ -78,7 +78,7 @@ static __always_inline int handle_ipv4(struct xdp_md *xdp)
 {
 	void *data_end = (void *)(long)xdp->data_end;
 	void *data = (void *)(long)xdp->data;
-	struct iptnl_info *tnl;
+	struct iptnl_info tnl;
 	struct ethhdr *new_eth;
 	struct ethhdr *old_eth;
 	struct iphdr *iph = data + sizeof(struct ethhdr);
@@ -102,10 +102,16 @@ static __always_inline int handle_ipv4(struct xdp_md *xdp)
 	vip.dport = dport;
 	payload_len = ntohs(iph->tot_len);
 
+
+	tnl.daddr.v4 = iph->daddr;
+	tnl.saddr.v4 = iph->saddr;
+	tnl.dmac = old_eth->h_dest;
+	#if 0
 	tnl = bpf_map_lookup_elem(&vip2tnl, &vip);
 	/* It only does v4-in-v4 */
 	if (!tnl || tnl->family != AF_INET)
 		return XDP_PASS;
+	#endif
 
 	/* The vip key is found.  Add an IP header and send it out */
